src/D2L.CodeStyle.TestAnalyzers/TestCaseData/TestCaseDataAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.TestAnalyzers.TestCaseData {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class TestCaseDataAnalyzer : DiagnosticAnalyzer {
+		public const string DiagnosticId = "D2L0010";
+		private const string Category = "Safety";
+
+		private const string Title = "Ensure test does not contain named property 'Throws' in TestCaseData.";
+		private const string Description = "Named property 'Throws' should not be used in TestCaseData.";
+		internal const string MessageFormat = "Use Assert.Throws or Assert.That in your test case instead of 'Throws' in TestCaseData for NUnit 3 compatibility";
+
+		private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			DiagnosticId,
+			Title,
+			MessageFormat,
+			Category,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true,
+			description: Description
+		);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create( Rule );
+
+		public override void Initialize( AnalysisContext context ) {
+			context.RegisterSyntaxNodeAction(
+				AnalyzeSyntaxNode,
+				SyntaxKind.InvocationExpression
+			);
+		}
+
+		private void AnalyzeSyntaxNode( SyntaxNodeAnalysisContext context ) {

```

Comment:
Tested a little bit, for `var symbol = context.SemanticModel.GetSymbolInfo( root )`, the symbols for the InvocationExpressions of the tests in `TestCaseDataAnalyzerTests.cs` are always `null`?